lua/barbar/render.lua

```diff
@@ -640,27 +478,33 @@ local function get_bufferline_group_clumps(layout, bufnrs, refocus)
         jump_letter.text = '  '
       end
     elseif icons_option.filetype.enabled then
-      local iconChar, iconHl = icons.get_icon(bufnr, activity)
-      local hlName = (activity == 'Inactive' and not config.options.highlight_inactive_file_icons)
+      local iconChar, iconHl = icons.get_icon(bufnr, activity_name)
+      local hlName = (activity_name == 'Inactive' and not config.options.highlight_inactive_file_icons)
         and 'BufferInactive'
         or iconHl
 
       icon.hl = icons_option.filetype.custom_colors and
-        hl_tabline('Buffer' .. activity .. 'Icon') or
-        (hlName and hl_tabline(hlName) or buffer_hl)
+        wrap_hl('Buffer' .. activity_name .. 'Icon') or
+        (hlName and wrap_hl(hlName) or buffer_hl)
       icon.text = #name.text > 0 and iconChar .. ' ' or iconChar
     end
 
-    --- The separator
-    --- @type barbar.render.group
+    local hl_sign = wrap_hl('Buffer' .. activity_name .. 'Sign')
+
     local left_separator = {
-      hl = clickable .. hl_tabline('Buffer' .. activity .. 'Sign'),
+      hl = clickable .. hl_sign,
       text = icons_option.separator.left,
     }
 
-    local pad = pinned and pinned_padding or padding
+    local pad_width = pinned and config.options.minimum_padding or layout.buffers.padding
+    local pad = {
+      hl = hl_sign,
+      text = (' '):rep(pad_width),
```

Comment:
Doing this allocates a new string every loop, whereas leaving it on line 523 allows these two particular parts of the barbar to be precalculated before the loop even starts.